Hardsuit Module Tweaks

ghostsheet opened the pull request:
🆑
rscadd: Hardsuit overlay icons will now change depending on what module is selected
tweak: Hardsuit drill mount now functions like a normal drill, being able to mine multiple tiles at once
tweak: Hardsuit drill mount energy usage has increase to a non insignificant amount
rscadd: Mounted plasma cutter has a close range mode to enable object interaction accessible by configuring
bugfix: Selecting a hardsuit module now checks for energy and suit status
bugfix: The energy blade projector can now fire darts again
rscadd: Mounted plasma cutter now has a sprite
rscadd: Mounted energy gun and Tazer now has a sprite
rscadd: Mounted medical injector now has a sprite
rscadd: Mounted grenade launchers now has a sprite
/🆑

This PR should closes #24583.

Perhaps I should add tech origin to more of modules while I'm here.

Preview of sprites:
https://i.imgur.com/U6I43Ig.gif

Summary of changes:
•Removes redundant spawns(1) operators, that did nothing, similar to this statement.

•Break down engage() proc into check() proc and then engage() which calls upon the check() proc. To allow for checks without engaging the modules.
•Added a select() proc which occurs whenever a hardsuit modules is selected. It updates the hardsuit to overlay sprites and have the selected module sprite show up, also includes a check() proc to stop people from selecting things while their suit has no power and such. 
•Made a special exception for deselecting all modules via rig verb, made it do the same checks, since I can't call a proc on nothing. [can't do null.select()]

 •Re-added and edited multiple overlay sprites of various modules back in. For the following modules: Energy gun/Tazer (same sprite), plasma cutter, medical injector, grenade launcher.
•Made them functional by correctly labelling the modules to the overlay sprites. Had to set  suit_overlay_active

•Added Plasma Cutter close range mode, allowing for object interaction with plasma cutter like slicing through un-powered airlocks and surgery for those maniacs. Basically spliced in, the device modules functions, which switches between device and gun mode dependant on it's "active" variable.(Perhaps I should make a new variable for it).  

•Removed the mounted drill's module cooldown delay, allowing it work like a normal drill, mining multiple tiles. While increasing its power usage to 1 Wh (a normal cell is 1000 Wh).

•Made the laser cannon it's own path. To prevent it stopping other mounted modules to be installed.
•Also change all the objects the the old laser cannon path with its new path.

•Made a new path for the mounted ninja crossbow, removing its safety while adding back its energy cost. Fixing a bug where the energy blade projectors can't shoot darts as said in description, as it uses the ninja crossbow which comes default with safeties on.

PsiOmegaDelta commented on code/modules/clothing/spacesuits/rig/modules/modules.dm:
```diff
@@ -213,6 +219,30 @@
+				holder.update_icon()
+
+	holder.selected_module = src
+	spawn(1)
```
Why this spawn?

PsiOmegaDelta replied:
Or any of the other spawns for that matter.

ghostsheet replied:
I really don't know, I was just follow the convention the last person did. lemme test it with out the spawn delay

ghostsheet replied:
Good point, it was all completely redundant as far as I can tell

ghostsheet commented:
Ready for review

ghostsheet commented:
I think the travis fail is unrelated? it is literally the same thing as before with a proper commit name.

ghostsheet committed:
Hardsuit Select module tweak

Selecting module now update icons
Mounted plasma cutter now has an object interaction mode
Mounted drill has no cooldown and an increased power cost
Split engage() proc to a check() and removed redundant

ghostsheet committed:
Mounted modules changes

All mounted gun will now show up with the shoulder mounted sprite (with the exception of energy blade projector)
Fixed energy blade projector, allowing it to shoot the projectile it was meant to
Made the lascannon into it's own type or path or whatever you call it, so it doesn't bug with other mounted module when installing into a suit

ghostsheet committed:
Sprites Update

Edited and readd a bunch of unused sprites back into the game.
Add my own personal plasma cutter module!
Fixed some inconsitent coding

ghostsheet committed:
Clean up code a bit more
